File: ruby/morse_code.rb
def encode(number)
  number.to_s.split("").map { |digit|  "-----.....----"[digit.to_i .. digit.to_i + 4] }.join
end

def decode(code)
  result = []
  (0 ... code.length).step(5) { |index|
    result.push("-----.....----".index(code[index, 5]))
  }
  result.join.to_i
end

puts encode 1203
puts decode "-----.....-----.....-----"

File: lua/morse_code.lua
local key = "-----.....----"

function encode(number)
   local result = ""
   for digit in string.gmatch(tostring(number), "%d") do
      local formatted_digit = tonumber(digit)
      result = result .. string.sub(key, formatted_digit, formatted_digit + 5)
   end
   return result
end

function decode(code)
   local result = ""
   for index = 1, #code, 5 do
      local digit = string.find(key, string.sub(code, index, index + 4), 1, true)
      if not digit then break end
      result = result .. (digit - 1)
   end
   return tonumber(result)
end

print(encode(1203))
print(decode("-----.....-----.....-----"))
